klippa/src/main.rs

```diff
@@ -100,6 +121,34 @@ fn populate_gids(gid_str: &str) -> Result<BTreeSet<GlyphId>, InvalidInputError>
             result.insert(GlyphId::new(glyph_id));
         }
     }
-    result.insert(GlyphId::new(0_u16));
+    Ok(result)
+}
+
+fn parse_unicodes(unicode_str: &Option<String>) -> Result<IntSet<u32>, InvalidInputError> {
+    let mut result = IntSet::empty();
+    if unicode_str.is_none() {
+        return Ok(result);
+    }
+    let re = regex::Regex::new(r"[<+->{},;&#\\xXuUnNiI\n\t\v\f\r]").unwrap();
+    let s = re.replace_all(unicode_str.as_deref().unwrap(), " ");
+    for cp in s.split_whitespace() {
+        if let Some((start, end)) = cp.split_once('-') {
+            let start: u32 = start
+                .parse::<u32>()
+                .map_err(|_| InvalidInputError::InvalidUnicode(start.to_owned().clone()))?;
```

Comment:
either one of `to_owned()` or `clone()` should be enough? if going from `&str` to `String` use `to_owned()`.